shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp

```diff
@@ -0,0 +1,122 @@
+
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::TransactionResponse::ResponseVariantType>(
+      ar, which > last ? last : which);
+  ;
+}
+
+namespace shared_model {
+  namespace proto {
+    /**
+     * TransactionResponse is a status of transaction in system
+     */
+    class TransactionResponse final : public interface::TransactionResponse {
+     private:
+      /// PolymorphicWrapper shortcut type
+      template <typename... Value>
+      using wrap = boost::variant<detail::PolymorphicWrapper<Value>...>;
+
+      /// lazy variant shortcut
+      using LazyVariantType = detail::LazyInitializer<ResponseVariantType>;
+
+      using RefTxResponse =
+          detail::ReferenceHolder<iroha::protocol::ToriiResponse>;
+
+     public:
+      /// Type of variant, that handle all concrete tx responses in the system
+      using ProtoResponseVariantType = wrap<StatelessFailedTxResponse,
+                                            StatelessValidTxResponse,
+                                            StatefulFailedTxResponse,
+                                            StatefulValidTxResponse,
+                                            CommittedTxResponse,
+                                            UnknownTxResponse>;
+
+      /// Type with list of types in ResponseVariantType
+      using ProtoResponseListType = ProtoResponseVariantType::types;
+
+      explicit TransactionResponse(
+          const iroha::protocol::ToriiResponse &command)
+          : TransactionResponse(RefTxResponse(command)) {}
+
+      explicit TransactionResponse(iroha::protocol::ToriiResponse &&command)
+          : TransactionResponse(RefTxResponse(std::move(command))) {}
+
+      /**
+       * @return hash of corresponding transaction
+       */
+      const interface::Transaction::HashType &transactionHash() const override {
+        return hash_;
+      };
+
+      /**
+       * @return attached concrete tx response
+       */
+      const ResponseVariantType &get() const override {
+        return *variant_;
+      }
+
+      ModelType *copy() const override {
+        return new TransactionResponse(
+            iroha::protocol::ToriiResponse(*response_));
+      }
+
+     private:
+      explicit TransactionResponse(RefTxResponse &&ref)
+          : response_(std::move(ref)),
+            variant_(detail::makeLazyInitializer([this] {
+              return ResponseVariantType(
+                  load<ProtoResponseListType>(*response_));
+            })),
+            // fixme @l4l: proto should be changed and this one replaced as well
+            //             or some other solution needed
+            hash_("") {}
+
+      // ------------------------------| fields |-------------------------------
+
+      // proto
+      RefTxResponse response_;
+
+      // lazy
+      LazyVariantType variant_;
+
+      // stub hash
+      crypto::Hash hash_;
```

Comment:
Use `Crypyo::Hash` instead and also wrap it with `Lazy`.

Reply:
There's no `crypto::Hash`